libs/contrib/Data/Telescope/Telescope.idr

```diff
@@ -0,0 +1,55 @@
+||| The telescope data structure.
+|||
+||| Indexing telescopes by their length (hopefully) helps inform the
+||| type-checker during inference.
+module Data.Telescope.Telescope
+
+import Data.Vect
+import Data.HVect
+import Syntax.PreorderReasoning
+
+-- Is there a way to make the definitions below positive?
+-- %default total
+
+
+
+infixl 4 -.,=.
+prefix 3 -|
+
+mutual
+  ||| A left-nested sequence of dependent types
+  public export
+  data Telescope : (k : Nat) -> Type where
+    Nil  : Telescope 0
+    (-.) : (gamma : Telescope k) -> (ty : TypeIn gamma) -> Telescope (S k)
+    
+  ||| A type with dependencies in the given context
+  public export
+  TypeIn : Telescope k -> Type
+  TypeIn gamma = (env : Environment gamma) -> Type
+  
+  ||| A tuple of values of each type in the telescope
+  public export
+  data Environment : Telescope k -> Type where
+    Empty : Environment []
+    (=.)  : Environment gamma -> ty env -> Environment (gamma -. ty)
```

Comment:
```suggestion
    (=.)  : (env : Environment gamma) -> ty env -> Environment (gamma -. ty)
```

I am actually surprised anything works given this issue?!

Reply:
Oh wow, thanks! I'm surprised as h/well. Another inconsistency?